### PR TITLE
Implement templating within containerTemplate.

### PR DESCRIPTION
**Pull request by EliZucker**
# Changes
_Taken from my commit message:_

Before, ApplyReplacements() would apply templating replacements to the containers within spec.Steps, but never to the single container at spec.ContainerTemplate. Note that this didn't work earlier because containerTemplate is merged with the container specs after templating is applied.

Isolate the container-specific templating in applyContainerReplacements()
and apply that logic to the containerTemplate field.

This addresses #811.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing) - **I don't think there is documentation to add here?**
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
`containerTemplate` now supports variable interpolation
```

I modeled the note above from [this similar PR](https://github.com/tektoncd/pipeline/pull/982) for consistency.

### CODE REVIEW

File: pkg/reconciler/v1alpha1/taskrun/resources/apply.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ limitations under the License.
 package resources
 
 import (
+	corev1 "k8s.io/api/core/v1"
```
Comment (vdemeester) [inline pkg/reconciler/v1alpha1/taskrun/resources/apply.go]: nit: This import needs to be above :angel:

File: pkg/reconciler/v1alpha1/taskrun/resources/apply.go [hunk 2]
```diff
@@ -60,42 +62,12 @@ func ApplyReplacements(spec *v1alpha1.TaskSpec, replacements map[string]string)
...
-			steps[i].VolumeMounts[iv].Name = templating.ApplyReplacements(v.Name, replacements)
-			steps[i].VolumeMounts[iv].MountPath = templating.ApplyReplacements(v.MountPath, replacements)
-			steps[i].VolumeMounts[iv].SubPath = templating.ApplyReplacements(v.SubPath, replacements)
-		}
+		applyContainerReplacements(&steps[i], replacements)
+	}
+
+	// Apply variable expansion to containerTemplate fields.
+	if spec.ContainerTemplate != nil {
+		applyContainerReplacements(spec.ContainerTemplate, replacements)
```
Comment (bobcatfish) [inline pkg/reconciler/v1alpha1/taskrun/resources/apply.go]: I just realized we need to do this for `stepTemplate` as well @EliZucker  😅 

lemme know if you don't want to do it, I can open a bug also

Comment (EliZucker) [reply pkg/reconciler/v1alpha1/taskrun/resources/apply.go]: See [this PR](https://github.com/tektoncd/pipeline/pull/1061)!

Comment (bobcatfish) [reply pkg/reconciler/v1alpha1/taskrun/resources/apply.go]: whoa so fast! thanks @EliZucker :D

**Pull request merged**